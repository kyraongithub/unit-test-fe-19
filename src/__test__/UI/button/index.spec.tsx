import Button from '@/components/UI/button';
import { fireEvent, render } from '@testing-library/react';

describe('Button Component', () => {
  it('render button component', () => {
    const { container } = render(<Button onclick={() => {}}>test</Button>);
    expect(container).toMatchSnapshot();
  });
  it('handle button click', () => {
    const clickFunction = jest.fn();
    const { getByText } = render(<Button onclick={clickFunction}>test</Button>);
    fireEvent.click(getByText('test'));
    expect(clickFunction).toHaveBeenCalled();
  });
});
