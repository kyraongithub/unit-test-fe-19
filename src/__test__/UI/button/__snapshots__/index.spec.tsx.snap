// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component render button component 1`] = `
<div>
  <button>
    test
  </button>
</div>
`;
